src/analysis/normed_space/units.lean

```diff
@@ -72,11 +78,187 @@ begin
+begin
+  simp only [is_O_iff, normed_field.norm_one, mul_one],
+  cases subsingleton_or_nontrivial R; resetI,
+  { refine ⟨1, eventually_of_forall (λ t, _)⟩,
+    have : ∥inverse (↑x + t)∥ = 0 := by simp,
+    linarith },
+  { cases is_O_iff.mp (@inverse_one_sub_norm R _ _) with C hC,
+    use C * ∥((x⁻¹:units R):R)∥,
+    have hzero : tendsto (λ t, - (↑x⁻¹ : R) * t) (𝓝 0) (𝓝 0),
+    { convert ((mul_left_continuous (-↑x⁻¹ : R)).tendsto 0).comp tendsto_id,
+      simp },
+    refine (inverse_add x).mp ((hzero.eventually hC).mp (eventually_of_forall _)),
+    intros t bound iden,
+    rw iden,
+    simp at bound,
+    have hmul := norm_mul_le (inverse (1 + ↑x⁻¹ * t)) ↑x⁻¹,
+    nlinarith [norm_nonneg (↑x⁻¹ : R)] }
+end
+
+/-- The function
+`λ t, inverse (x + t) - (∑ i in range n, (- x⁻¹ * t) ^ i) * x⁻¹`
+is `O(t ^ n)` as `t → 0`. -/
+lemma inverse_add_norm_diff_nth_order (x : units R) (n : ℕ) :
+  is_O (λ (t : R), inverse (↑x + t) - (∑ i in range n, (- ↑x⁻¹ * t) ^ i) * ↑x⁻¹)
+  (λ t, ∥t∥ ^ n) (𝓝 (0:R)) :=
+begin
+  by_cases h : n = 0,
+  { simpa [h] using inverse_add_norm x },
+  have hn : 0 < n := nat.pos_of_ne_zero h,
+  simp [is_O_iff],
+  cases (is_O_iff.mp (inverse_add_norm x)) with C hC,
+  use C * ∥(1:ℝ)∥ * ∥(↑x⁻¹ : R)∥ ^ n,
+  have h : eventually_eq (𝓝 (0:R))
+    (λ t, inverse (↑x + t) - (∑ i in range n, (- ↑x⁻¹ * t) ^ i) * ↑x⁻¹)
+    (λ t, ((- ↑x⁻¹ * t) ^ n) * inverse (x + t)),
+  { refine (inverse_add_nth_order x n).mp (eventually_of_forall _),
+    intros t ht,
+    convert congr_arg (λ a, a - (range n).sum (pow (-↑x⁻¹ * t)) * ↑x⁻¹) ht,
+    simp },
+  refine h.mp (hC.mp (eventually_of_forall _)),
+  intros t _ hLHS,
+  simp only [neg_mul_eq_neg_mul_symm] at hLHS,
+  rw hLHS,
+  refine le_trans (norm_mul_le _ _ ) _,
+  have h' : ∥(-(↑x⁻¹ * t)) ^ n∥ ≤ ∥(↑x⁻¹ : R)∥ ^ n * ∥t∥ ^ n,
+  { calc ∥(-(↑x⁻¹ * t)) ^ n∥ ≤ ∥(-(↑x⁻¹ * t))∥ ^ n : norm_pow_le _ hn
+    ... = ∥↑x⁻¹ * t∥ ^ n : by rw norm_neg
+    ... ≤ (∥(↑x⁻¹ : R)∥ * ∥t∥) ^ n : _
+    ... =  ∥(↑x⁻¹ : R)∥ ^ n * ∥t∥ ^ n : mul_pow _ _ n,
+    exact pow_le_pow_of_le_left (norm_nonneg _) (norm_mul_le ↑x⁻¹  t) n },
+  have h'' : 0 ≤ ∥(↑x⁻¹ : R)∥ ^ n * ∥t∥ ^ n,
+  { refine mul_nonneg _ _;
+    exact pow_nonneg (norm_nonneg _) n },
+  nlinarith [norm_nonneg (inverse (↑x + t))],
+end
+
+/-- The function `λ t, inverse (x + t) - x⁻¹` is `O(t)` as `t → 0`. -/
+lemma inverse_add_norm_diff_first_order (x : units R) :
+  is_O (λ t, inverse (↑x + t) - ↑x⁻¹) (λ t, ∥t∥) (𝓝 (0:R)) :=
+by { convert inverse_add_norm_diff_nth_order x 1; simp }
+
+/-- The function
+`λ t, inverse (x + t) - x⁻¹ + x⁻¹ * t * x⁻¹`
+is `O(t ^ 2)` as `t → 0`. -/
+lemma inverse_add_norm_diff_second_order (x : units R) :
+  is_O (λ t, inverse (↑x + t) - ↑x⁻¹ + ↑x⁻¹ * t * ↑x⁻¹) (λ t, ∥t∥ ^ 2) (𝓝 (0:R)) :=
+begin
+  convert inverse_add_norm_diff_nth_order x 2,
+  ext t,
+  simp only [range_succ, range_one, sum_insert, mem_singleton, sum_singleton, not_false_iff,
+    one_ne_zero, pow_zero, add_mul],
+  abel,
+  simp
+end
+
```

Comment:
Could you state the continuity of inverse at any unit? It follows directly from what you have done above, and it does not follow in general from differentiability as differentiability only makes sense when you have more structure.